src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -191,6 +192,7 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
         type_to_fuse_map empty_fuse_map = {};
         manager.register_pass<ov::pass::Validate>();
         //  call ConvertPrecision with keep_precision_sensitive_in_fp32 = true
+        manager.register_pass<ov::pass::SoftmaxFusion>();
```

Comment:
Why you add this transformation for GPU pipeline? Is it not added in common pipeline?

Reply:
By default it was called inside `MOCTransformations`, and `MOCTransformations` were called inside `CommonOptimizations`, but `CommonOptimizations` are called in the end of pipeline. Therefore I called it here in advance to get all softmax pattern fused before the `ConvertPrecision` is run.

Do you think we should add a comment why we call this fusing in advance?

Reply:
Yes, please add comment. But still, is it really needed? My expectation was that MOC pipeline was run before GPU pipeline. In `convert_model` or `read_model`

Reply:
Honestly, looks like WA here. Can we have the same situation for other fusion transformations which are affected by `ConvertPrecision`? May we should have more general approach to solve such problems like ignoring some `Convert` nodes in the patterns? Or can you remind why `ConvertPrecision` is called after `CommonOptimizations`? 

fyi, @slyalin, @vladimir-paramuzov

Reply:
We call `ConvertPrecision` before `CommonOptimization` so that pattern matcher is not broken. Because if we call CommonOptimization earlier then on a bunch of operation there will be Cost+Decompression Convert instead of a single Const. @vladimir-paramuzov confirmed that personally.

So it looks like this w/a is inevitable.

Reply:
@itikhono, please share your opinion.

@mvafin, `read_model` doesn't call MOC. This is one of the desired things -- to call MOC in read_model -- from consistency perspective to align convert_model and read_model behaviors. But, as @mateusztabaka showed in his measurements, we cannot do it immediately without significantly regressing in read_time or without significant investments in the reoragnization of the transformation pipeline.

But even if we call it in read_model, the current convention is that during compile_model we shouldn't rely on that, because we don't really know where the model come from and whether it was modified manually. So it can require a full optimization pipeline to be applied.

ConvertPrecision is called in that place because "Const+Decompress patterns will be instead of just Consts and some pattern matchers potentially could not match". Just because nobody knowns how many transformation should be modified to support Const+Decompress we are calling it in this way. It looks to be much bigger topic that requires more dedicated analysis among all the transformation. Nobody committed to investigate part.

Looks like we only this transformation called twice. So this doesn't look to have major impact. In comparison to the bigger task described above, this is only thing that we can do considering amount of resources involved.

But the picture actually involves more details from plugins side, especially for big models where we are trying to minimize touching big weights in the pipeline. Should be discussed offline.

Reply:
Also, why don't we call it for CPU plugin? It seems to be also affected.

Reply:
Ok, let's leave it at that for now.
Could you create a ticket to delete this call and mention the ticket number in this comment?